sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_update(progress_bar_t *progress_bar)
+{
+    assert(progress_bar->value < 101);
+
+    /* Hide cursor */
+    printf("\033[?25l");
+
+    /* Move cursor at the beginning of the line */
+    putchar('\r');
+
+    /* Display progress bar prefix if any */
+    if (progress_bar->prefix[0]) {
+        printf("%s", progress_bar->prefix);
+    }
+
+    printf(PROGRESS_BAR_PREFIX_CHARACTER);
```

Comment:
here and below: why not using character literals and putchar?

Reply:
For consistency with the _full_ and _empty_ characters that could contain unicode characters. The fact that the variables ends with `CHARACTER` is confusing indeed but it's supposed to be a single character and can be ASCI or Unicode.

Reply:
Nice, so I can make leet `--<[xxx   ]>--` bars! :)